crates/solidity/inputs/language/src/lib.rs

```diff
@@ -1,24 +1,28 @@
-//! This crate is responsible for compiling the language definition and exposing it to downstream crates.
-//!
-//! While it's possible to directly compile the language, we do it here once, to ensure that:
-//! 1. Expensive parsing and validation is done only once.
-//! 2. Errors are reported only once, instead of repeating for every crate.
-//!
-//! Call the [`SolidityLanguageExtensions::load_solidity`] method to load the precompiled language definition.
+pub mod dsl;
+pub use dsl::*;
+
+use std::path::PathBuf;
+
 use anyhow::Result;
+use cargo_emit::rerun_if_changed;
 use codegen_schema::types::{LanguageDefinition, LanguageDefinitionRef};
+use infra_utils::paths::PathExtensions;
 
 pub trait SolidityLanguageExtensions {
-    /// Loads the precompiled Solidity language definition.
     fn load_solidity() -> Result<LanguageDefinitionRef>;
 }
 
-// Set by the build script.
-static LANGUAGE_DEFINITION_BIN: &'static str = env!("COMPILED_SOLIDITY_LANGUAGE_DEFINITION_BIN");
-
 impl SolidityLanguageExtensions for LanguageDefinition {
+    /// We compile the language definition only once, and then expose it here to all downstream crates.
+    /// This ensures that:
+    /// 1. Expensive parsing and validation is done only once.
+    /// 2. Errors are reported only once, instead of repeating for every crate.
     fn load_solidity() -> Result<LanguageDefinitionRef> {
-        let buffer = std::fs::read(LANGUAGE_DEFINITION_BIN)?;
+        let bin_file_path = PathBuf::from(env!("COMPILED_SOLIDITY_LANGUAGE_DEFINITION_BIN"));
+
+        rerun_if_changed!(bin_file_path.unwrap_str());
```

Comment:
Looks like this reverts the changes from #561
Is this intentional?

Reply:
I think there is something wierd going in #565 -> #561

Reply:
But I reverted this to #565

Reply:
> I think there is something wierd going in #565 -> #561

What do you mean by weird, exactly?

The reason for this change is that Cargo markers are only meant for build scripts and it's not necessary to emit that directive for downstream users, since Cargo will understand that if a build script target is marked as dirty, so will be the library target, and thus every library consumer (that pulls this library for `load_solidity`), so it's not necessary.

This change works if we add
```rust
rerun_if_changed!(bin_file_path.unwrap_str());
```
for the build script, i.e. in crates/solidity/inputs/language/build.rs, see https://github.com/NomicFoundation/slang/pull/561/files#diff-a0461e3fd0b7fd3afe069982f7fabdb0b015dd015833114eb5532372bf96d987R23

Reply:
I couldn't grok the merge result

Reply:
But I think I know what was going wrong. Fixed